### ISSUE TITLE
Travis failure

### ISSUE THREAD

**Issue by alexcarsello** [StanfordAHA/CGRAGenerator#96]
Just pushed some changes to dev to restructure and improve my tests for the global controller, and they caused the travis tests to fail. Here's the Travis log: https://travis-ci.org/StanfordAHA/CGRAGenerator/builds/406477158

The changes don't seem to be related to anything I changed... anyone have any ideas on what could be causing this?

**Comment by leonardt** [comment]
weird, it looks like it's failing for some PE unit tests, but I see that your changes don't touch any part of that. I'm trying to reproduce the issue locally but can't, going to investigate further. It doesn't seem to be an intermittent issue given that running another build still causes the crash.

I might push some commits to poke around the travis test

**Comment by leonardt** [comment]
The last dev build I can find was 16 days ago, which matches the commit history (the commit before your last) and it seemed to pass, https://travis-ci.org/StanfordAHA/CGRAGenerator/builds/400997607.

Could be that one of the dependencies has changed? The only thing I see are `https://github.com/phanrahan/pe` which hasn't changed since April. `https://github.com/leonardt/bit_vector` had a change in last 2 weeks, so that could be related. I'm going to investigate that.

**Comment by leonardt** [comment]
I pushed a new branch with the previous commit that passed, and the build is going to fail (https://travis-ci.org/StanfordAHA/CGRAGenerator/builds/406933871) so it looks like this isn't related to your changes. Haven't figured anything else out yet.

**Comment by leonardt** [comment]
Haven't been able to figure this one out yet, for now, to unblock yourself you could disable the pe tests temporarily in a new a branch (e.g. `gc-tests`) and make the following change. Ideally we keep `dev` failing so we don't forget to resolve this issue.

```diff
diff --git a/.travis.yml b/.travis.yml
index 5109913..d1fb18c 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -55,7 +55,7 @@ script:
   - export GENESIS_HOME=/tmp/Genesis2/Genesis2Tools
   - export PATH=$GENESIS_HOME/bin:$GENESIS_HOME/gui/bin:$PATH
   - export PERL5LIB=$GENESIS_HOME/PerlLibs/ExtrasForOldPerlDistributions:$PERL5LIB
-  - ./run_tests.sh
+  # - ./run_tests.sh
 
 
   # Redo serpent summary maybe
```

**Comment by steveri** [comment]
On a hunch, I tried running master for the first time in ?26 days?
Yes it fails too.  Not good!   :(
Also: I though master was supposed to run every night automatically?

https://travis-ci.org/StanfordAHA/CGRAGenerator/builds/407184455

**Comment by steveri** [comment]
@leonardt My magic eight-ball says signs point to 'bit_vector' ... did you try regressing it to say a month ago and see what happens?

**Comment by leonardt** [comment]
I think `CGRAFlow` runs nightly, but that was not setup for this repo. I'll try reverting to an older bit_vector version and see what happens

**Comment by leonardt** [comment]
Magic eight-ball was correct! For this build: https://travis-ci.org/StanfordAHA/CGRAGenerator/builds/407257871 I fixed the bit_vector version to `0.24a` and it passed.  So, somewhere between 0.24a and 0.26a there was a regression. I'll keep investigating, but others can merge or rebase on master to get the working bit_vector version for now.

**Comment by alexcarsello** [comment]
Thanks, Lenny!

**Comment by leonardt** [comment]
Okay, found the issue. The latest bit_vector version updates how the `signed` flag is passed on when performing operations. The `pe` spec made an assumption about the old behavior that introduced a bug when the new behavior was released.

I've updated the `pe` spec to fix the issue: https://github.com/phanrahan/pe/commit/67cab7ae574eb54c3f78014946d71832b8c631c1

The fix worked in my debug branch: https://travis-ci.org/StanfordAHA/CGRAGenerator/builds/407332581

Merged the changes into master, so I think things should be back in order. Will also merge them into dev so that build is working again.
